File: app/src/main/java/com/addon/AddonUtils.java
package com.addon;

import me.peace.utils.LogUtils;

public class AddonUtils {
    public static double div(double a,double b){
        return a / b;
    }

    public static double div(int a,int b){
        return a / b;
    }

    public static void stack(){
        StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
        for (StackTraceElement element : stackTraceElements){
            LogUtils.e("stack",
                element.getClassName() + "," + element.getMethodName() + "," + element.getLineNumber() + "," + element.getFileName());
        }
    }
}

File: app/src/main/java/com/addon/AddonUtils.kt
package com.addon

import me.peace.utils.LogUtils.e

object AddonUtils {
    fun div(a: Double, b: Double): Double {
        return a / b
    }

    fun div(a: Int, b: Int): Double {
        return (a / b).toDouble()
    }

    fun stack() {
        val stackTraceElements = Thread.currentThread().stackTrace
        for (element in stackTraceElements) {
            e(
                "stack",
                element.className + "," + element.methodName + "," + element.lineNumber + "," + element.fileName
            )
        }
    }
}